### DISCUSSION TITLE
Usage Binding from Rest API Datasource in Automation.

### DISCUSSION THREAD

**Discussion by aborniak** [Budibase/budibase#10776] [Help] [answered]
Hello Guys, 
I can't figure out how to use Binding from Rest API Datasource in Automation. 
Example: 
Rest API Datasorce "BambooHR Users API" has Query "Employee Table" where I have two Bindings:
1. since_date
2. employee_table_date
<img width="1390" alt="Screenshot 2023-06-02 at 17 01 35" src="https://github.com/Budibase/budibase/assets/49722408/672616cc-c695-4001-8150-1b3fa5eb872d">
I have the following "Params" there: 
<img width="1001" alt="Screenshot 2023-06-02 at 17 06 03" src="https://github.com/Budibase/budibase/assets/49722408/57a6c0ff-181a-43c7-978a-56afa5c7d698">

I have an EXTERNAL DATA CONNECTOR in Automation that makes a request to the "BambooHR Users API" -> "Employee Table" API. 
<img width="679" alt="Screenshot 2023-06-02 at 17 08 03" src="https://github.com/Budibase/budibase/assets/49722408/d4823003-6c05-4275-9bdf-0e98adf55b9b">

I have QUERY ROWS steps that make requests to the internal Budibase database and I want to use the Filter  date >= employee_table_date (Binding value that has been used in Step 1 ^).
<img width="1236" alt="Screenshot 2023-06-02 at 17 19 47" src="https://github.com/Budibase/budibase/assets/49722408/917da786-70d4-42f8-8ba3-0336b21dd918">
**Questions:** 
<img width="1251" alt="Screenshot 2023-06-02 at 17 23 52" src="https://github.com/Budibase/budibase/assets/49722408/0235492d-95fb-4c89-a1aa-6d1f80587b9b">
1. How can I get the Employee_table_date  value from Step 1 in JavaScript and Handlebars ^?
2. I have some code in JS SCRIPTING and don't know if printing the values of some variables in Test Details Output is possible.
<img width="1131" alt="Screenshot 2023-06-02 at 17 33 01" src="https://github.com/Budibase/budibase/assets/49722408/55f95892-cdf6-4163-ac74-f6e987200507">
<img width="709" alt="Screenshot 2023-06-02 at 17 45 08" src="https://github.com/Budibase/budibase/assets/1437571/c41823c8-e803-4149-8a40-7c18409bb3f0">
Thank you!

**Comment by aborniak** [comment]
Guys,
Do you have any suggestions/tips/questions?

**Reply by melohagan** [reply, marked as answer]
Hey @aborniak 

Sorry for the delayed response, let me have a go at answering some of your questions.

> 1. How can I get the Employee_table_date value from Step 1 in JavaScript and Handlebars ^?

Employee_table_date is an **input** field for Step 1, so you don't need to 'get' the value because you already know what it is?

> 2. I have some code in JS SCRIPTING and don't know if printing the values of some variables in Test Details Output is possible.

You can try `console.log("Last changed: ", data.lastChanged)` or the **Backend log** step.

**Reply by aborniak** [reply]
Hey @melohagan  
Thank you!
